test/ffmpeg-qsv/transcode/transcoder.py

```diff
@@ -173,14 +220,20 @@ def transcode(self):
 
     for n, output in enumerate(self.outputs):
       for channel in xrange(output.get("channels", 1)):
+        width = output.get("width", 0)
+        height = output.get("height", 0)
         encoded = self.goutputs[n][channel]
         yuv = get_media()._test_artifact(
           "{}_{}_{}.yuv".format(self.case, n, channel))
-        call(
-          "ffmpeg -i {} -pix_fmt yuv420p -vframes {}"
-          " -y {}".format(encoded, self.frames, yuv))
+        if width > 0 and height > 0:
+          self.scale_to_yuv(yuv, encoded, output["codec"], output["mode"])
+        else:
+          call(
+            "ffmpeg -i {} -pix_fmt yuv420p -vframes {}"
+            " -y {}".format(encoded, self.frames, yuv))
```

Comment:
Why decode to yuv based on "mode" if only scaling is involved, yet always sw decode when scaling is not involved.